### PR TITLE
feat: refactor part of bash script to be included in TF

### LINKED ISSUE #84: Move EKS autoscaling to TF
Currently autoscaling happens from a set of scripts and files, while there is also https://github.com/terraform-aws-modules/terraform-aws-eks/blob/4898bf326d7019fdd9cc33b533a226d6ef3a0f52/examples/irsa_autoscale_refresh/charts.tf#L9-L78 . Let's migrate to it!

### PR DESCRIPTION
**Pull request by bendehaan**
Closes #84

### CODE REVIEW

File: aws/build-an-deploy-aws.sh [hunk 1]
```diff
@@ -45,12 +39,13 @@ esac
 ACCOUNT_ID=$(aws sts get-caller-identity | jq '.Account' -r)
 echo "ACCOUNT_ID=${ACCOUNT_ID}"
 
+CLUSTERNAME="$(terraform output -raw cluster_name)"
+STATE_BUCKET="$(terraform output -raw state_bucket_name)"
+IRSA_ROLE_ARN="$(terraform output -raw irsa_role_arn)"
+EBS_ROLE_ARN="$(terraform output -raw ebs_role_arn)"
```
Comment (commjoen) [inline aws/build-an-deploy-aws.sh:46]: can you please echo the configuration here? (e.g. the variables set) so we don't run into debug issues :) ?

File: aws/README.md [hunk 2]
```diff
@@ -34,27 +34,25 @@ terraform apply
...
 
 **Note-II**: The cluster you create has its access bound to the public IP of the creator. In other words: the cluster you create with this code has its access bound to your public IP-address if you apply it locally.
 
 1. export your AWS credentials (`export AWS_PROFILE=awsuser`)
-2. check whether you have the right profile by doing `aws sts get-caller-identity` and make sure you have enough rights with the caller its identity and that the actual accountnumber displayed is the account designated for you to apply this TF to.
-3. Do `terraform init` (if required, use tfenv to select TF 0.13.1 or higher )
-4. The bucket ARN will be asked for in the next 2 steps. Take the one provided to you and add `arn:aws:s3:::` to the start. e.g. ``arn:aws:s3:::terraform-20230102231352749300000001`
+2. check whether you have the right profile by doing `aws sts get-caller-identity`. Make sure you have the right account and have the rights to do this.
+3. Do `terraform init` (if required, use tfenv to select TF 0.14.0 or higher )
+4. The bucket ARN will be asked in the next 2 steps. Take the one provided to you in the output earlier.
```
Comment (commjoen) [inline aws/README.md:46]: can we still have a nice arn example of what people need to look for here?

File: aws/main.tf [hunk 3]
```diff
@@ -131,3 +137,51 @@ module "eks" {
...
+  role_name                              = "wrongsecrets-load-balancer-controller"
+  attach_load_balancer_controller_policy = true
+
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:aws-load-balancer-controller"]
+    }
+  }
+}
```
Comment (commjoen) [inline aws/main.tf:187]: YES !!! 3 STACKS OMITTED! LOVE THIS!

File: aws/build-an-deploy-aws.sh (hunk 1 above)
Comment (bendehaan) [reply aws/build-an-deploy-aws.sh:46]: Done

**Review by commjoen — APPROVED** [review]
Thank you @bendehaan , this makes it all a lot cleaner and less hacky based on cloudformation stacks!

**Pull request merged**